### PR TITLE
`set-labels` feature change, handle slice, change log

### PR DESCRIPTION
**Pull request by zyy98**
fixed previous way of handling slice, add more path with slice and refactor that part of code after discussion with Yuwen.
change log to show general information to avoid being truncated.

### CODE REVIEW

File: functions/go/set-labels/transformer/labels.go
Comment (yuwenma) [inline functions/go/set-labels/transformer/labels.go:101]: func Transform() {
     // filter unrelated resources. like localconfig or Kptfile metaconfig
     err := func() {
       // set meta labels
       UpdateLabelsInMeta
       // set pod labels 
       UpdatePodLabels 
       // labelSelector
       ...
       // optional (Not CreateifNot exist)  
     }()
     if err != nil {
         LogError
     }
     LogSummary
}

File: functions/go/set-labels/transformer/labels.go
Comment (yuwenma) [inline functions/go/set-labels/transformer/labels.go:257]: super nit: can we use the switch/case format?

**Comment by yuwenma** [comment]
The new format is much better! I think it's a good starting point to help us improving the SDK and make the `set-labels` code shorter and more readable.
I'd like to get another review from @natasha41575

File: functions/go/set-labels/transformer/labels.go
Comment (natasha41575) [inline functions/go/set-labels/transformer/labels.go:27]: nit: this seems a bit verbose, suggest the following instead:

```
	resultCount := make(map[string]int)
	return &LabelTransformer {
		ResultCount: resultCount,
	}
```

File: functions/go/set-labels/transformer/labels.go
Comment (natasha41575) [inline functions/go/set-labels/transformer/labels.go:23]:
```suggestion
func NewLabelTransformer() *LabelTransformer {
```

File: functions/go/set-labels/transformer/labels.go
Comment (natasha41575) [inline functions/go/set-labels/transformer/labels.go:87]: nit: I think these comments (L79, L83, and L87) are not necessary, the name of the methods are descriptive enough

File: functions/go/set-labels/transformer/labels.go
Comment (natasha41575) [inline functions/go/set-labels/transformer/labels.go:118]: I think having this check inside the `setJobTemplateSpecMeta` is a little bit confusing, the name of the method `setJobTemplateSpecMeta` seems to imply that we already know that the object is a Job. 

I think a better structure would be to change lines 90-99 above to something like

```
if hasJobTempalteSpec(o) {
  p.setJobTemplateSpecMeta
}
```

and same for `p.setVolumeClaimTemplates(o)` and `p.setNetworkPolicyRule(o)`

File: functions/go/set-labels/transformer/labels.go
Comment (natasha41575) [inline functions/go/set-labels/transformer/labels.go:257]: Discussion point for @yuwenma, I think this would be a good use case for getting this information from the openapi data instead of having to hardcode all of these things.

File: functions/go/set-labels/transformer/labels.go
Comment (zyy98) [reply functions/go/set-labels/transformer/labels.go]: I see. I left this change because for functions like `setSelector`, I need to check if it has different selector type inside the function. So for consistency I just left all `hasXXX` function inside `setXXX`. But for the corner cases, it seems a good idea to separate them out.

File: functions/go/set-labels/transformer/labels.go
Comment (natasha41575) [inline functions/go/set-labels/transformer/labels.go:397]: nit: suggest renaming 'element' to 'labelCountValue' or something similar

File: functions/go/set-labels/transformer/labels.go
Comment (natasha41575) [inline functions/go/set-labels/transformer/labels.go:398]:
```suggestion
		msg := fmt.Sprintf("set labels {%v} %v times", key, element)
```

**Review by natasha41575 — APPROVED** [review]
Code structure LGTM, defer to @yuwenma for final approval

**Review by yuwenma — APPROVED** [review]

**Comment by yuwenma** [comment]
This PR looks great! This is a good starting point to let us continue improving the SDK and adding features. like the OpenApi schema as @natasha41575 suggested.

**Pull request merged**
